vtkAddonConfig.cmake.in

```diff
@@ -1,3 +1,5 @@
+include("${CMAKE_CURRENT_BINARY_DIR}/vtkAddonTargets.cmake")
```

Comment:
I think this would throw an error when vtkAddon is built as part of Slicer since the targets are just exported to SlicerTargets.cmake.

Reply:
In that case, we should do the following in Slicer:

```diff
diff --git a/Libs/CMakeLists.txt b/Libs/CMakeLists.txt
index 0b3ff1bc0d..5e80b64ec7 100644
--- a/Libs/CMakeLists.txt
+++ b/Libs/CMakeLists.txt
@@ -96,18 +96,19 @@ foreach(dir ${dirs})
   set(${dir_noslash}_INSTALL_LIB_DIR ${Slicer_INSTALL_LIB_DIR})
   set(${dir_noslash}_INSTALL_SHARE_DIR ${Slicer_INSTALL_ROOT}share/${dir_noslash})
   set(${dir_noslash}_INSTALL_NO_DEVELOPMENT ${Slicer_INSTALL_NO_DEVELOPMENT})
-  set(${dir_noslash}_EXPORT_FILE ${Slicer_BINARY_DIR}/SlicerTargets.cmake)
   set(${dir_noslash}_FOLDER Core-Libs/${dir_noslash})
 
   # Check if library sources were specified as a remote module
   if(DEFINED ${dir}_SOURCE_DIR AND Slicer_BUILD_${dir})
     message(STATUS "Configuring library: ${dir} [${dir}_SOURCE_DIR: ${${dir}_SOURCE_DIR}]")
+    set(${dir_noslash}_EXPORT_FILE ${Slicer_BINARY_DIR}/SlicerTargets.cmake)
     add_subdirectory(
       ${${dir}_SOURCE_DIR}
       ${CMAKE_CURRENT_BINARY_DIR}/${dir}
       )
   else()
     message(STATUS "Configuring library: ${dir}")
+    set(${dir_noslash}_EXPORT_FILE ${Slicer_BINARY_DIR}/SlicerTargets.cmake)
     add_subdirectory(${dir})
   endif()

```

Reply:
Neverind, the logic I implemented in Libs/CMakeLists.txt is need to be improved I think.

Reply:
Here it is:

```
diff --git a/Libs/CMakeLists.txt b/Libs/CMakeLists.txt
index 0b3ff1bc0d..867eb52596 100644
--- a/Libs/CMakeLists.txt
+++ b/Libs/CMakeLists.txt
@@ -100,12 +100,15 @@ foreach(dir ${dirs})
   set(${dir_noslash}_FOLDER Core-Libs/${dir_noslash})
 
   # Check if library sources were specified as a remote module
-  if(DEFINED ${dir}_SOURCE_DIR AND Slicer_BUILD_${dir})
-    message(STATUS "Configuring library: ${dir} [${dir}_SOURCE_DIR: ${${dir}_SOURCE_DIR}]")
-    add_subdirectory(
-      ${${dir}_SOURCE_DIR}
-      ${CMAKE_CURRENT_BINARY_DIR}/${dir}
-      )
+  if(DEFINED Slicer_BUILD_${dir})
+    # Then, add directory  only if library is expected to be built by Slicer
+    if(Slicer_BUILD_${dir})
+      message(STATUS "Configuring library: ${dir} [${dir}_SOURCE_DIR: ${${dir}_SOURCE_DIR}]")
+      add_subdirectory(
+        ${${dir}_SOURCE_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}/${dir}
+        )
+    endif()
   else()
     message(STATUS "Configuring library: ${dir}")
     add_subdirectory(${dir})
```